assets/training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -268,9 +276,23 @@ def download_model(self):
             )
 
 
-def download_model(model_source: str, model_id: str, download_dir: Path):
+def download_model(model_source: str, model_id: str, download_dir: Path, token: str):
     """Download model and return model information."""
     if model_source == ModelSource.HUGGING_FACE.value:
+        if token:
+            try:
```

Comment:
lets move this logic to respective downloader?

Reply:
I guess, we should not keep it inside downloader. Right now it seems like we need it for downloading only, but in future we might have different use cases. We should keep download class independent of it.
Thoughts?

Reply:
Could you share an example where this might be needed outside?

Reply:
So token is basically a way to authenticate user for using any service of HF like transformers, datasets, inference API.

Reply:
If we want to have model info, from HF API in that case as well we will be requiring this token.

Reply:
Ideally we can have one HFclass, and there we can use token..and then other subclass downloads, model_info_gathering all these gonna use that token authentication

Reply:
> If we want to have model info, from HF API in that case as well we will be requiring this token.

make sense lets create a hugging face util for this. Right now, even model info fetching is part of common util. Can be done as part of sep. PR